src/odemis/driver/piezomotor.py

```diff
@@ -593,9 +599,12 @@ def isMovingClosedLoop(self, axis):
         if d2 & 0b100:
             raise PMDError(6, "External limit reached.")
```

Comment:
Adding the current positioning a debug message could help. Especially if the motor has two limits (lower and higher) since then you can find out on which end of the limit the motor currently is.

Reply:
good one, I'll also add on which axis the limit was reached